packages/ember-htmlbars/lib/hooks/create-fresh-scope.js

```diff
@@ -46,13 +49,124 @@
   the current view's `controller`.
 */
 
+function Scope(parent) {
+  this._self = null;
```

Comment:
my vote would be for these to be `undefined`, as `null` to me implies no value. But actually this exists merely for shaping.